components/caption_images/src/main.py

```diff
@@ -80,35 +107,21 @@ def transform(
         device = "cuda" if torch.cuda.is_available() else "cpu"
         logger.info(f"Device: {device}")
 
-        processor = AutoProcessor.from_pretrained(model_id)
+        processor = BlipProcessor.from_pretrained(model_id)
         model = BlipForConditionalGeneration.from_pretrained(model_id)
 
-        # load and transform the images
-        images = dataframe["images_data"]
-        loaded_images = [load(image) for image in images]
-        transformed_images = [
-            transform(image, processor, device) for image in loaded_images
-        ]
-
-        # batch images together
-        batches = [
-            collate(batch)
-            for batch in toolz.partition_all(batch_size, transformed_images)
-        ]
-
-        # caption images
-        delayed_model = dask.delayed(model.to(device))
-        captions = [
-            caption(batch, delayed_model, processor, max_new_tokens)
-            for batch in batches
-        ]
-
-        # join lists into a single Dask delayed object
-        captions = flatten(captions)
-        delayed_series = dd.from_delayed(captions, meta=pd.Series(dtype="str"))
-        captions_df = delayed_series.to_frame(name="captions_text")
-
-        return captions_df
+        dataframe = dataframe["images_data"].map_partitions(
+            caption_images,
+            model=model,
+            processor=processor,
+            batch_size=batch_size,
+            max_new_tokens=max_new_tokens,
+            device=device,
+            meta={0: str}
+        )
+        dataframe.columns = ["captions_text"]
```

Comment:
Could you follow a similar for defining the `meta` dataframe as you did in the the laion component
```
meta_df = pd.DataFrame(columns=["id", "captions_text"], dtype="string")
meta_df = meta_df.set_index("id")

        dataframe = dataframe["images_data"].map_partitions(
            caption_images,
            model=model,
            processor=processor,
            batch_size=batch_size,
            max_new_tokens=max_new_tokens,
            device=device,
            meta=meta_df
        )
```
Haven't tested this but I think it makes it a bit more explicit what dataframe you're returning.

Reply:
I'm not doing this here since I didn't want to set the column names inside the `caption_images`. I kept those names on the "component level", since that's where they are specified in the specification as well.